tms/src/main/java/com/netcracker/edu/tms/service/ProjectServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.netcracker.edu.tms.service;
+
+import com.netcracker.edu.tms.dao.ProjectDao;
+import com.netcracker.edu.tms.model.Project;
+import com.netcracker.edu.tms.model.Task;
+import com.netcracker.edu.tms.model.User;
+import com.netcracker.edu.tms.model.UsersToProjects;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigInteger;
+import java.util.List;
+
+@Service
+public class ProjectServiceImpl implements ProjectService {
```

Comment:
Let's add test for this class using mockito

The main idea is mocking (some kind of stubbing) dao layer to isolate service layer

Example:

package com.netcracker.edu.tms.service;

import com.netcracker.edu.tms.dao.ProjectDao;
import com.netcracker.edu.tms.model.Project;
import org.junit.Assert;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.Mockito;

import java.math.BigInteger;

import static org.mockito.Mockito.mock;

public class ProjectServiceImplTest {

    private static final String TEST_NAME = "test";

    // here we specify that projectDao is mocked
    @Mock
    private ProjectDao projectDao = mock(ProjectDao.class);

    private ProjectService projectService = new ProjectServiceImpl(projectDao);

    @Test
    public void getProjectById() throws Exception {
        Project expected = new Project(BigInteger.ONE, BigInteger.ONE, TEST_NAME);

       // here we tell to projectDao returning what we want
        Mockito.when(projectDao.getProjectById(expected.getId())).thenReturn(expected);

        Project actual = projectService.getProjectById(expected.getId());
        Assert.assertEquals(expected, actual);
    }

}

You may say it is stupid to test method which contains just one line
    public Project getProjectById(BigInteger projectId) {
        return projectDao.getProjectById(projectId);
    }

But if method has complex logic it makes sense

I ask for this task just for educational purpose and introducing to Mockito testing gramework
Do not be lazy to visit https://site.mockito.org/